File: 169-majority-element/majority-element.py
class Solution:
    def majorityElement(self, nums: List[int]) -> int:
        count = 0
        element = nums[0]

        for i in range(0,len(nums)):
            if(count == 0):
                count = 1
                element = nums[i]
            elif(element == nums[i]):
                count += 1
            else:
                count -= 1

        return element
        

File: 169-majority-element/majority-element.ts
function majorityElement(nums: number[]): number {
    let count:number = 0;
    let element:number = 0;

    for(let i:number = 0;i<nums.length;i++)
    {
        if(count == 0)
        {
            count = 1;
            element = nums[i];
        }
        else if(element == nums[i])
        {
            count++;
        }
        else
        {
            count--;
        }
    }

    return element;
    
};